src/test/java/api/requests/scenarios/HoldShelfExpirationDateTests.java

```diff
@@ -93,16 +98,59 @@ void requestWithShelfExpirationDateForSpExpiryInHoursAndMinutes(
 
-    assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+    ZonedDateTime zdtWithZoneOffset  = ZonedDateTime.parse(storedRequest.getString("holdShelfExpirationDate"),
+      DateTimeFormatter.ISO_ZONED_DATE_TIME);
+
+    if (isSameDay(interval, amount, zdtWithZoneOffset)) {
+
+      assertHoldShelfExpirationDateBasedOnStrategy(storedRequest,servicePoint,amount,interval);
+   }
+    else {
+      assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
       storedRequest.getString("holdShelfExpirationDate"),
       isEquivalentTo(interval.addTo(ClockUtil.getZonedDateTime(), amount)));
+    }
+  }
+
+  private boolean isSameDay(ChronoUnit interval, int amount, ZonedDateTime zdtWithZoneOffset) {
+    return interval.addTo(getZonedDateTime(), amount).toLocalDate().equals(LocalDate.now(zdtWithZoneOffset.getZone()));
+  }
+
+  private void assertHoldShelfExpirationDateBasedOnStrategy(JsonObject storedRequest, String servicePoint, int amount, ChronoUnit interval) {
+    switch (servicePoint) {
+      case "cd4" : assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(atEndOfDay(interval.addTo(ClockUtil.getZonedDateTime(), amount)).minusDays(1)));
+        break;
+      case "cd5" : assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(ClockUtil.getZonedDateTime().plusMinutes(42)));
+        break;
+      case "cd6": assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(ClockUtil.getZonedDateTime().plusDays(2).with(LocalTime. of(23, 59, 59).plusSeconds(1))));
+        break;
+      case "cd7" : assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(ClockUtil.getZonedDateTime().plusDays(1).with(LocalTime. of(5, 0, 0))));
+        break;
+      case "cd8" : assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(ClockUtil.getZonedDateTime().plusDays(1).with(LocalTime. of(0, 10, 0))));
+        break;
+      case "cd9" :
+      case "cd10" : assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(atEndOfDay(interval.addTo(ClockUtil.getZonedDateTime(), amount)).plusDays(1)));
+        break;
+    }
```

Comment:
In case `cd6` the result of this call `ClockUtil.getZonedDateTime().plusDays(2).with(LocalTime.of(23, 59, 59).plusSeconds(1))` is a time object that represents  "the end of the current service point hours" because that is the strategy that belongs to `cd6`.

How could this code and other cases here make that relationship more clear? How do I know that that time object represents "end of the current service point hours" without really studying the `ServicePointExamples`?

We could  assign `ClockUtil.getZonedDateTime().plusDays(2).with(LocalTime.of(23, 59, 59).plusSeconds(1))` to a variable called `endOfCurrentServicePointHours`. And do the same for the other items checked in `isEquivalentTo`. That would make it a bit easier to follow.

But I'm also a bit unclear here (in the case of `cd6` but also `cd7`and 'cd8`) why 2 days plus one day minus a second is actually the _end of the current service point hours_. Why not some other time value? What is determining the 3 days - 1 second? It seems like this time value is  being hardcoded based on a calendar that I'm not seeing. Like why don't we have something like `getServicePoint{"cd6").closingTime()` to get the "end of the current service point hours" and compare that here for example?

Reply:
done

Reply:
Maybe before you merge add a comment before `daysToAddCd7` and `daysToAddCd8` that these time spans reflect existing strategies for closed library dates used to calculate due dates and that we _think_ they also apply here.

Reply:
> Maybe before you merge add a comment before `daysToAddCd7` and `daysToAddCd8` that these time spans reflect existing strategies for closed library dates used to calculate due dates and that we _think_ they also apply here.

Still having doubts about this. Can't these two variable names be more descriptive? Isn't each date one of the 6 strategies we have in the enum? Which strategy is each? Shouldn't that be the variable name. And if not, isn't there something wrong with the test or with the class that we are using itself?

Reply:
With fresh eyes this looks a little more clear to me, but seems like `daysToAddCd7` and `daysToAddCd8` could be defined once since they have the same value. Also couldn't the variable name be a bit more semantically meaningful like maybe `spanToNextServicePointHours` (maybe you have a better name). I _think_ that's what the duration is right?